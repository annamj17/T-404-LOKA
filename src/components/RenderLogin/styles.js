import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  screens: {
    flex: 1,
    flexDirection: 'column',
    height: '100%'
  },
  logoImage: {
    width: 370,
    height: 280,
    marginTop: 'auto',
    alignSelf: 'center'
  },
  loginHeader: {
    fontSize: 25,
    color: 'black',
    textAlign: 'center'
  },
  inputView: {
    width: '80%',
    alignSelf: 'center',
    borderRadius: 10,
    borderWidth: 2,
    height: 50,
    marginBottom: 10,
    marginTop: 20,
    justifyContent: 'center',
    padding: 20
  },
  inputText: {
    height: 50,
    color: 'black'
  },
  btn: {
    flexDirection: 'row',
    width: 90,
    height: 40,
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 10,
    marginTop: 20,
    marginLeft: 40,
    backgroundColor: '#D3D3D3'
  },
  imageContainer: {
    marginTop: 'auto',
    height: '12%'
  },
  errorText: {
    color: 'red',
    textAlign: 'center'
  }
});
