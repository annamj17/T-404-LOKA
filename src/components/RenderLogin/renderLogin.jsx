import React, { useState } from 'react';
import {
  View,
  Text,
  Image,
  TextInput,
  TouchableOpacity,
  ActivityIndicator,
  KeyboardAvoidingView
} from 'react-native';

import { logIn } from '../../services/authentication.service';
import styles from './styles';
import logoImage from '../../../assets/K3imaginelogo.png';
import bottomImage from '../../../assets/Capture.png';

const RenderLogin = ({ navigation }) => {
  const [userName, setUserName] = useState('');
  const [password, setPassword] = useState('');
  const [isLoading, setIsLoading] = useState(false);
  const [error, setError] = useState(false);

  return (
    <View style={styles.screens}>
      <KeyboardAvoidingView behavior="padding">
        <Image style={styles.logoImage} source={logoImage} resizeMode="contain" />
        <Text style={styles.loginHeader}>Welcome, please log in to your account</Text>
        <View style={styles.inputView}>
          <TextInput
            value={userName}
            name="userName"
            type="text"
            onChangeText={text => setUserName(text)}
            style={styles.inputText}
            placeholder="Username *"
            textContentType="username"
          />
        </View>
        <View style={styles.inputView}>
          <TextInput
            value={password}
            name="password"
            type="password"
            onChangeText={text => setPassword(text)}
            style={styles.inputText}
            placeholder="Password *"
            textContentType="password"
            secureTextEntry
          />
        </View>
        {error && <Text style={styles.errorText}>There was an error, please try again</Text>}
        <TouchableOpacity
          style={styles.btn}
          name="login"
          onPress={async () => {
            setIsLoading(true);
            const loggedIn = await logIn(userName, password);
            setIsLoading(false);
            if (loggedIn) {
              setError(false);
              navigation.navigate('App');
            } else {
              setError(true);
            }
          }}
        >
          <Text> Login </Text>
          <ActivityIndicator animating={isLoading} size="small" />
        </TouchableOpacity>
      </KeyboardAvoidingView>
      <Image style={styles.imageContainer} source={bottomImage} resizeMode="stretch" />
    </View>
  );
};

export default RenderLogin;
