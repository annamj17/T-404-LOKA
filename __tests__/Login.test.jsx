import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import RenderLogin from '../src/components/RenderLogin/renderLogin';

describe('Should render the Login page without crashing', () => {
  it('renders correctly', () => {
    const rendered = renderer.create(<RenderLogin />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});

describe('Login tests', () => {
  const onNavigateSpy = jest.fn();
  const props = {
    navigation: {
      navigate: onNavigateSpy
    }
  };

  const isLoading = false;
  const setIsLoading = false;
  const userName = '';
  const setUserName = '';

  const wrapper = shallow(
    <RenderLogin {...props} {...isLoading} {...setIsLoading} {...userName} {...setUserName} />
  );

  it('Should find 2 TextInputs', () => {
    expect(wrapper.find('TextInput')).toHaveLength(2);
  });

  it('Should find the TextInput for userName', () => {
    expect(wrapper.find('TextInput[type="text"]')).toHaveLength(1);
  });

  it('Should find the TextInput for password', () => {
    expect(wrapper.find('TextInput[type="password"]')).toHaveLength(1);
  });

  it('Should handle onPress for login', () => {
    const wrap = mount(<RenderLogin {...props} />);
    wrap
      .find('TouchableOpacity[name="login"]')
      .first()
      .props()
      .onPress();
    expect(userName).toBeFalsy();
  });
});
