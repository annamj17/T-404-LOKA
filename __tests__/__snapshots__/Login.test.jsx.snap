// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the Login page without crashing renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "height": "100%",
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "paddingBottom": 0,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        Object {
          "alignSelf": "center",
          "height": 280,
          "marginTop": "auto",
          "width": 370,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 25,
          "textAlign": "center",
        }
      }
    >
      Welcome, please log in to your account
    </Text>
    <View
      style={
        Object {
          "alignSelf": "center",
          "borderRadius": 10,
          "borderWidth": 2,
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginTop": 20,
          "padding": 20,
          "width": "80%",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        name="userName"
        onChangeText={[Function]}
        placeholder="Username *"
        rejectResponderTermination={true}
        style={
          Object {
            "color": "black",
            "height": 50,
          }
        }
        textContentType="username"
        type="text"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "borderRadius": 10,
          "borderWidth": 2,
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginTop": 20,
          "padding": 20,
          "width": "80%",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        name="password"
        onChangeText={[Function]}
        placeholder="Password *"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "color": "black",
            "height": 50,
          }
        }
        textContentType="password"
        type="password"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D3D3D3",
          "borderRadius": 10,
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "center",
          "marginLeft": 40,
          "marginTop": 20,
          "opacity": 1,
          "width": 90,
        }
      }
    >
      <Text>
         Login 
      </Text>
      <ActivityIndicator
        animating={false}
        color="#999999"
        hidesWhenStopped={true}
        size="small"
      />
    </View>
  </View>
  <Image
    resizeMode="stretch"
    source={1}
    style={
      Object {
        "height": "12%",
        "marginTop": "auto",
      }
    }
  />
</View>
`;
